README.md

```diff
@@ -516,29 +516,225 @@ func main() {
 ```
 </details>
 
+
+<details>
+<summary>Generate Embeddings</summary>
+
+```go
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"fmt"
+	"os"
+
+	"github.com/sashabaranov/go-openai"
+)
+
+func getEmbedding(ctx context.Context, client *openai.Client, input []string) ([]float32, error) {
+
+	resp, err := client.CreateEmbeddings(ctx, openai.EmbeddingRequest{
+		Input: input,
+		Model: openai.AdaEmbeddingV2,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Data[0].Embedding, nil
+}
+
+func main() {
+
+	ctx := context.Background()
+	client := openai.NewClient("your token")
+
+	// example selections
+	selections := []string{
+		"Welcome to the go-openai interface, which will be the gateway for golang software engineers to enter the OpenAI development world.",
+		"It was tasty and fresh. The other one I bought was old and tasted moldy. But this one was good.",
+		"Great coffee at a good price. I'm a subscription buyer and I buy this month after month. What more can I say?",
+		"This chocolate is amazing..I love the taste and smell, this is the only chocolate for me...I found a new love!",
+		"I love this coffee!  And such a great price.  Will buy more when I am running out which will be soon.",
+		"The Raspberry Tea Syrup is great. I can use it for hot and cold drinks as well in certain recipes.",
+		"Everyone that dips with this loves it!  So easy to use! Olive oil and tasty bread is all you need.",
+		"This is a favorite of mine for using over ice. Even bought it to give out as Christmas gifts last year.",
+		"If you like a great , hot, sauce then buy this. If spicy with heat isn't to your liking then don't buy it.",
+		"My name is Aceld, and I am a Golang software development engineer. I like young and beautiful girls.",
+		"The competition was held over two days,24 July and 2 August. The qualifying round was the first day with the apparatus final on the second day.",
+		"There are 4 types of gymnastics apparatus: floor, vault, pommel horse, and rings. The apparatus final is a competition between the top 8 gymnasts in each apparatus.",
+	}
+
+	// Generate embeddings
+	var selectionsEmbeddings [][]float32
+	for _, selection := range selections {
+		embedding, err := getEmbedding(ctx, client, []string{selection})
+		if err != nil {
+			fmt.Printf("GetEmedding error: %v\n", err)
+			return
+		}
+		selectionsEmbeddings = append(selectionsEmbeddings, embedding)
+	}
+
+	// Write embeddings binary data to file
+	file, err := os.Create("embeddings.bin")
+	if err != nil {
+		fmt.Printf("Create file error: %v\n", err)
+		return
+	}
+	defer file.Close()
+
+	encoder := gob.NewEncoder(file)
+	err = encoder.Encode(selectionsEmbeddings)
+	if err != nil {
+		fmt.Printf("Encode error: %v\n", err)
+		return
+	}
```

Comment:
I think file I/O and marshalling is largely out of scope to the purpose of this example. Could you please remove it?

Reply:
@sashabaranov  Sure, you're right. Do you have any suggestions on how to store vector data more efficiently? I would appreciate some advice.

Reply:
@aceld I think the storage of vector data is largely is out of scope for this README — the point is just to show an example, not to build vector-search DB.